streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBWindowStore.java

```diff
@@ -48,11 +49,12 @@
 
     private static final long USE_CURRENT_TIMESTAMP = -1L;
 
-    private static class Segment extends RocksDBStore<byte[], byte[]> {
+    // use the ByteBuffer wrapper for underlying rocksDB keys since they are used for hashing data structures
+    private static class Segment extends RocksDBStore<ByteBuffer, byte[]> {
```

Comment:
The `ByteBuffer` wrapper here will only be used for serde, and no `get*` method will be triggered on the keys. I am a bit on the fence to have all `ByteArraySerde`, `ByteBufferSerde` and `BytesSerde` in the serialization package, but if people feel strong about it I can add them (I actually have these functions in local repo, but was not sure to commit them).